From fc9a444d57a1508d49b19b14045f801754ad6fe2 Mon Sep 17 00:00:00 2001
From: Artur Wieczorek <artwik@wp.pl>
Date: Thu, 19 Mar 2015 22:42:44 +0100
Subject: [PATCH 2/2] Don't mix signed/unsigned integers in arithmetic
 operations when normalizing wxDC scale factors.

Variable holding GCD value should be of the same type as variables holding devExt and logExt to avoid wrong results of /= operation when dividend is a negative value.
---
 src/msw/dc.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/msw/dc.cpp b/src/msw/dc.cpp
index 58d8960..d22e7c0 100644
--- a/src/msw/dc.cpp
+++ b/src/msw/dc.cpp
@@ -2012,7 +2012,7 @@ void wxMSWDCImpl::RealizeScaleAndOrigin()
     // In GDI anisotropic mode only devExt/logExt ratio is important
     // so we can reduce the fractions to avoid large numbers
     // which could cause arithmetic overflows inside Win API.
-    unsigned int gcd = wxGCD(abs(devExtX), abs(logExtX));
+    int gcd = wxGCD(abs(devExtX), abs(logExtX));
     devExtX /= gcd;
     logExtX /= gcd;
     gcd = wxGCD(abs(devExtY), abs(logExtY));
-- 
1.9.5.msysgit.0

